platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/Bucket.java

```diff
@@ -241,8 +240,14 @@ public void putValue(final K key, final long value) {
                     bucketData.position(result.entryOffset());
                     bucketData.writeBytes(remainder);
                 }
+                if (bucketIndexFieldOffset > result.entryOffset()) {
+                    // It should not happen with default implementation, but if buckets are serialized
+                    // using 3rd-party tools, field order may be arbitrary, and "bucket index" field
+                    // may be after the deleted entry
+                    bucketIndexFieldOffset -= result.entrySize();
+                }
```

Comment:
Seems like this code is making a very particular assumption about where exactly that offset is located. I guess this should be fine as long as we only use a limited number of "third-party tools."

Reply:
No particular assumptions. There used to be an assumption that "bucket index" field was always written first, but with this change the assumption is gone

Reply:
Sounds good.

Reply:
I'll updated it in the subsequent PR